### DISCUSSION TITLE
client.collection('users').confirmVerification(token) does not authenticate user?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#1036] [Q&A]
I am using `client.collection('users').confirmVerification(token)` on my custom frontend to verify accounts.

My expectation was that the user would be properly authenticated (authState updated) as part of this call, but that appears not to be the case. Is there a reason why using an account verification token should NOT also authenticate the user? If a user logs out, clears localStorage, is in incognito/private mode, or simply attempts verification from a different browser, the ux feels strange because it requires them to log in again.

This unexpected behavior is especially noticeable on mobile because clicking `Verify` from an email link will open an embedded browser rather than the browser used to initiate the verification request.

**Comment by ganigeorgiev** [comment]
Hm, I'll think a little more on this and may add an exception for the `confirm-verification` endpoint.

This has changed with the v0.8.0-rc3 release - https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc3.

The reason behind the change was that it was inconsistent with the other email confirm actions and previously it required returning 2 different success responses depending whether the auth collection has email auth enabled or not.

**Reply by ganigeorgiev** [reply]
@benallfree After some consideration I've decided to keep it as it is (aka. returning 204 and no token) for several reasons:

- it is consistent with the other `request-*` and `confirm-*` email actions
- it doesn't create a loophole and we don't have to return 2 success responses when there is no auth method enabled for the auth collection (eg. when you have disabled authentication for your application)
- allowing direct authentication only by clicking on the email link could eventually conflict with a custom multi-factor auth implementation (eg. when using the `OnRecordAuthRequest` hook or a custom middleware)

To be honest, I'm still not 100% sure that it is the right implementation because I understand and agree that this is not very UX friendly in some cases as you highlighted, but I also think that having only the auth methods (`authWithPassword`, `authWithOAuth2`, `authRefresh`) being responsible for the authentication is easier to reason, document and test (especially when only a single success response format is expected).
